batch/data-plane/src/main/java/com/microsoft/azure/batch/BatchClient.java

```diff
@@ -80,6 +97,18 @@ public static BatchClient open(BatchCredentials credentials) {
         return new BatchClient(credentials);
     }
 
+    /**
+     * Creates an instance of {@link BatchClient} associated with the specified restClient and baseurl.
+     *
+     * @param restClient A {@link RestClient} object specifying the definition of rest client used to
+     *                   interact with batch endpoint.
+     * @param baseUrl A  String object specifying the the batch end point.
+     * @return The new {@link BatchClient} instance.
+     */
+    public static BatchClient open(RestClient restClient, String baseUrl) {
+        return new BatchClient(restClient, baseUrl);
+    }
```

Comment:
Do we need new public API? This could probably be package-protected?

Reply:
Keyvault has a public client constructor with RestClient as a parameter.
Had this public to keep it consistent.
But changed it to protected and Batch team can choose to expose it later, if needed.

Reply:
I still don't understand why we need this new public API? Can it be package-protected (i.e. with no modifier and therefore not public API)?

Reply:
There seems to be github glitch.
I already changed it to protected

Reply:
Not protected, package-protected (with no modifier)

Reply:
you mean package-private  (no modifier) ?
https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html
Got confused.

Reply:
yes

Reply:
Updated to package-private.